overlord/snapstate/policy.go

```diff
@@ -31,3 +33,23 @@ func inUseFor(deviceCtx DeviceContext) func(snap.Type) (boot.InUseFunc, error) {
 		return boot.InUse(typ, deviceCtx)
 	}
 }
+
+func isCoreBootDevice(st *state.State) (bool, error) {
+	// TODO: turn this into a function that takes a DeviceContext
+	// that can be nil (TransitionCore case)
+
+	// Use device context just to find out if this is a core boot device.
+	// state.ErrNoState is returned if we cannot find the model,
+	// which can happen on classic when upgrading from ubuntu-core
+	// to core snap (model is set later).
+	deviceCtx, err := DeviceCtx(st, nil, nil)
+	if err != nil {
+		if !errors.Is(err, state.ErrNoState) {
+			return false, err
+		}
+		return false, nil
+	} else {
```

Comment:
(nitpick) I guess we could leave out the else here? Would feel a bit more natural (plus there is a stray newline too)